### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by richmccartney** [canonical-websites/vanillaframework.io#66]
Putting http://vanillaframework.io/ through pa11y there are 9 errors recorded which I feel need some attention.

To see these issues locally do `npm install -g pa11y` and then run `pa11y --ignore warning --ignore notice http://vanillaframework.io`

**Comment by richmccartney** [comment]
• Error: This link points to a named anchor "navigation-closed" within the document, but no anchor exists with that name.
   ├── WCAG2AA.Principle2.Guideline2_4.2_4_1.G1,G123,G124.NoSuchID
   ├── #navigation > div > div > a:nth-child(3)
   └── <a href="#navigation-closed" class="p-navigation__toggle--close" title="close menu">Menu</a>

 • Error: This link points to a named anchor "main-content" within the document, but no anchor exists with that name.
   ├── WCAG2AA.Principle2.Guideline2_4.2_4_1.G1,G123,G124.NoSuchID
   ├── #navigation > div > nav > span > a
   └── <a href="#main-content">Jump to main content</a>

 • Error: Img element missing an alt attribute. Use the alt attribute to specify a short text alternative.
   ├── WCAG2AA.Principle1.Guideline1_1.1_1_1.H37
   ├── html > body > div:nth-child(3) > div > div:nth-child(1) > div > div > img
   └── <img class="p-heading-icon__img" src="https://assets.ubuntu.com/v1/fe40dc6a-pictogram_lightweight01b-dark.svg">

 • Error: Img element missing an alt attribute. Use the alt attribute to specify a short text alternative.
   ├── WCAG2AA.Principle1.Guideline1_1.1_1_1.H37
   ├── html > body > div:nth-child(3) > div > div:nth-child(2) > div > div > img
   └── <img class="p-heading-icon__img" src="https://assets.ubuntu.com/v1/ba665746-pictogram_extension01-dark.svg">

 • Error: Img element missing an alt attribute. Use the alt attribute to specify a short text alternative.
   ├── WCAG2AA.Principle1.Guideline1_1.1_1_1.H37
   ├── html > body > div:nth-child(3) > div > div:nth-child(3) > div > div > img
   └── <img class="p-heading-icon__img" src="https://assets.ubuntu.com/v1/bcf3db83-pictogram_opensource01-dark.svg">

 • Error: This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.87:1. Recommendation: change text colour to #007aa4.
   ├── WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
   ├── #getting-started > div:nth-child(2) > div:nth-child(2) > pre > code > span:nth-child(1)
   └── <span class="hljs-built_in">export</span>

 • Error: This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.87:1. Recommendation: change text colour to #007aa4.
   ├── WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
   ├── #getting-started > div:nth-child(2) > div:nth-child(2) > pre > code > span:nth-child(2)
   └── <span class="hljs-built_in">pwd</span>

 • Error: This text input element does not have a name available to an accessibility API. Valid names are: label element, title attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputText.Name
   ├── #mc-embedded-subscribe-form > div:nth-child(1) > div > input
   └── <input type="text" name="b_56dac47c206ba0f58ec25f314_36f7d8394e" tabindex="-1" value="">

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #mc-embedded-subscribe-form > div:nth-child(1) > div > input
   └── <input type="text" name="b_56dac47c206ba0f58ec25f314_36f7d8394e" tabindex="-1" value="">

**Comment by barrymcgee** [comment]
Thanks @richmccartney - good tool!

- Missing `alt` tags have been addressed by https://github.com/canonical-websites/vanillaframework.io/pull/78

- The #navigation-closed error is a quirk of a pure-CSS nav. The header already has an id of "navigation".

- The Mailchimp input field errors related to the name (which we cannot change). This field is wrapped in a div with `aria-hidden` anyway.

- Color contrast issues in the code blocks are controlled by the plugin used so outside of the scope of this project.
